lab/goals.md

```diff
@@ -0,0 +1,30 @@
+# OPI Lab Goals
+
+The OPI members may contribute some hardware
+to be co-managed between the member and the lab staff.
+Having hardware in the OPI lab helps
+to validate compatibility and to avoid regressions.
+
+## Usage
+
+The OPI lab should be used for:
+
+- automatic testing
+
+  * scheduled functional validation
+  * patch validation (CI)
+
+- demonstration
+
+  * allowed users might reserve their own machine
+
+The data generated in the lab shall not be used
+for competitive marketing purposes.
+
+## Properties
+
+In order to serve the OPI community, these properties should be targetted:
+
+- reliability: always running, without false positive result
+- neutrality: same automatic tests and configurations on all HW/SW platforms
+- security: access is restricted to avoid unexpected changes
```

Comment:
Multi-tenant, so we can carve the resources up and share them amongst the CI systems and developers.

Reply:
I would like to hear people's thoughts on the OPI lab as a vehicle for independent certification.  The phrase "validate compatibility" is not as strong as "certification." Note that "certification" can simply mean whatever set of tests people have already have in place is agreed to by the TSC to belong to some official OPI test suite.

Reply:
I would add another property:

- ease-of-use: commonly-used services should be provided to increase adoption

Intent here is to increase adoption by not forcing end users to re-invent the wheel.  I would suggest to identify user profiles.  For each profile, identify services the profile will need.  We can then discuss whether it makes sense for the lab to provide commonly used services vs. users are on their own to provision.

Reply:
> I would like to hear people's thoughts on the OPI lab as a vehicle for independent certification. The phrase "validate compatibility" is not as strong as "certification." Note that "certification" can simply mean whatever set of tests people have already have in place is agreed to by the TSC to belong to some official OPI test suite.

Sure we could have certification, but for now there is no such program, so I'm reluctant to add it now.

Reply:
> I would add another property:
> 
>     * ease-of-use: commonly-used services should be provided to increase adoption
> 
> 
> Intent here is to increase adoption by not forcing end users to re-invent the wheel. I would suggest to identify user profiles. For each profile, identify services the profile will need. We can then discuss whether it makes sense for the lab to provide commonly used services vs. users are on their own to provision.

I think you are saying we should test the API with some well known clients.
I fully agree, we need clients to test with.
However, it should be in a separate test because a failure in this test may be a problem in the client, not OPI.

Reply:
How adding such test becomes a lab property?